### DISCUSSION TITLE
重新渲染画布数据，除了toJSON()和fromJSON()外，还有其他方法或实现思路吗？

### DISCUSSION THREAD

**Discussion by xiazhaohui** [antvis/X6#2232] [Q&A]
重新渲染画布数据，除了toJSON()和fromJSON()外，还有其他方法或实现思路吗？

节点稍微一多，toJSON()导出的数据量太大了，把生成的数据传给后端，后端再透传回前端，数据量动辄几十KB，前端再拿此数据使用fromJSON()渲染到画布。

数据量太大了，不合理，有没有除了fromJSON外其他的回显方法。

<img width="790" alt="image" src="https://user-images.githubusercontent.com/82207908/174233897-9acd8a25-ba40-4447-94d0-eccc620dff19.png">

**Comment by NewByVector** [comment]
你的问题更像是画布数据怎么和服务端交互，而不是怎么回显。例如你新增一个节点，你只要把最新数据提交到服务端就行了，不需要重新获取服务端数据再来刷新画布。如果要考虑异常情况，可以使用 undo 机制将添加操作回退。

**Reply by xiazhaohui** [reply]
场景不是那样的，画布里面有很多节点，比如流程A、流程B、流程C等等，关闭某一流程画布前将全部数据传给服务端保存。在再次打开某一个流程画布时，再从服务端拿到该流程的数据，来渲染到一个新画布中。上传和渲染现在是使用toJSON和fromJSON来实现的

**Reply by NewByVector** [reply]
可以考虑每次只传改动的内容到服务端。

**Reply by xiazhaohui** [reply]
> 可以考虑每次只传改动的内容到服务端。

这个思路确实是比较高效的，就是跟目前手里的需求有出入，无法按这样来实现
